llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp

```diff
@@ -4081,8 +4087,15 @@ bool SelectionDAGLegalize::ExpandNode(SDNode *Node) {
       // For PIC, the sequence is:
       // BRIND(load(Jumptable + index) + RelocBase)
       // RelocBase can be JumpTable, GOT or some sort of global base.
-      Addr = DAG.getNode(ISD::ADD, dl, PTy, Addr,
-                          TLI.getPICJumpTableRelocBase(Table, DAG));
+      if (!DAG.getTarget().shouldPreservePtrArith(
```

Comment:
Probably should have this in a more convenient property in the DAG

Reply:
Can't we use `getMemBasePlusOffset` here?